java/code/src/com/redhat/rhn/manager/content/ContentSyncManager.java

```diff
@@ -754,6 +759,18 @@ public void refreshRepositoriesAuthentication(
             if (tokenOpt.isPresent()) {
                 newAuth = new SCCRepositoryTokenAuth(tokenOpt.get());
             }
+            else if (repo.getProducts().isEmpty()) {
+                log.debug("Repo '" + repo.getUrl() + "' not in the product tree. Skipping");
+                continue;
+            }
```

Comment:
I'd personally prefer better not to have the `continue;` jump, but wrapping instead with the condition criteria, something like

```java
if (repo.getProducts().isEmpty(){
  log.debug("Repo '" + repo.getUrl() + "' not in the product tree. Skipping");
}
else {
  if (tokenOpt.isPresent()) {
    newAuth = new SCCRepositoryTokenAuth(tokenOpt.get());
  }
  else if {
    ....
  }
}
```

just from the logic perspective and the code readability, so I can understand the flow and the reasoning (if I understood the logic correctly).

Reply:
Well, I am a fan of not indent things too much.
In this case there is also another reason why we getter not call repo.getProducts() on all repos. This cause a single SQL query which result in performance issues when we do it on all of our repos. When we get repos from SCC with token auth I expect that we have them in the tree. Even if not, we do not more operations on them.

For the other we might trigger a head request which slow down things and with this I try everything to minimize the cases where we have to do it. But at thi splace it does not hurt as in the next "if" we call the same to find out if we need to do more checks. The SQL query will be done anyway. So we do not loose anything.

I think I will keep it as it is now.

Reply:
As long as there is a reason behind and it affects performances or whatever else, I'm just fine with it. Thanks for taking the time to explain it here. :+1: